httpcore/_sync/http11.py

```diff
@@ -127,14 +127,14 @@ def _send_request_body(self, request: Request) -> None:
             event = h11.Data(data=chunk)
             self._send_event(event, timeout=timeout)
 
-        event = h11.EndOfMessage()
-        self._send_event(event, timeout=timeout)
+        self._send_event(h11.EndOfMessage(), timeout=timeout)
 
     def _send_event(
-        self, event: H11Event, timeout: Optional[float] = None
+        self, event: h11.Event, timeout: Optional[float] = None
```

Comment:
I think we still want our send_event to have a specific allowlist of events we send:

```
H11SendEvent = Union[
    h11.Request,
    h11.Data,
    h11.EndOfMessage,
    h11.ConnectionClosed,
]
```

```suggestion
        self, event: H11SendEvent, timeout: Optional[float] = None
```

Reply:
👍 makes sense to me. Seems like we would need this with the [overload](https://github.com/encode/httpcore/pull/526/files#r901136483) for correct typing.